### DISCUSSION TITLE
Regarding ww3_grid.inp and subgrid information

### DISCUSSION THREAD

**Discussion by krishnanathira** [NOAA-EMC/WW3#774] [Q&A] [answered]
Hi all

I have created bathymetry files (domain.depth_ascii, domain.obsr_lev1, and domain.maskorig_ascii) using the latest version of Gridgen. Unfortunately, when I am running the ww3_grid, I am getting an error as follows,
```
Sub-grid information: Not available.

 *** WAVEWATCH III ERROR IN W3GRID : 
     ERROR IN READING FROM INPUT FILE
     IOSTAT =    0
```

Can you please help me to solve this?
Thanks in advance
Athira

**Comment by ukmo-ccbunney** [comment]
Can you supply your ww3_grid.nml file?
It might be worth checking the IDLA values are correct for your input files...

**Reply by krishnanathira** [reply]
Thanks a lot for your response.
Here's my ww3_grid.nml file

[ww3_grid.txt](https://github.com/NOAA-EMC/WW3/files/9457525/ww3_grid.txt)

Kind regards
Athira

**Comment by ukmo-ccbunney** [comment, marked as answer]
@krishnanathira 
In your `ww3_grid.txt` file, you appear to be setting grid values for regular, unstructured and SMC grid types...
Whilst this is not necessarily the root of your issue, it would be worth removing the sections from your `ww3_grid.nml` file that are not relevant to your regular rectangilar grid setup (which is what i assume you are using).

I.e. remove the &CURV_NML,  &UNSTR_NML and &SMC_NML sections. This will make the problem easier to diagnose.

The error you are getting suggests that one of your input files (depth, mask, etc) is failing to be read correctly.
You have IDLA set to 3 for you input files which means "read line by line from top to bottom" (see inp/ww3_grid.inp for details)
Are your input files formatted this way?

What input files are you providing?

**Reply by krishnanathira** [reply]
Thank you so much.
I will remove &CURV_NML, &UNSTR_NML and &SMC_NML sections and see.
Let me explain to you my inputs
I am using rectilinear grid setup (u10,v10) data from ERA5. 
Then gridgen outputs ie, domain.depth_ascii, domain.obsr_lev1, and domain.maskorig_ascii. 
Here are my bathymetry files, and grid. inp file attached. 
[ww3_grid.inp.txt](https://github.com/NOAA-EMC/WW3/files/9459988/ww3_grid.inp.txt)
[domain_depth_ascii.txt](https://github.com/NOAA-EMC/WW3/files/9459958/domain_depth_ascii.txt)
[domain_maskorig_ascii.txt](https://github.com/NOAA-EMC/WW3/files/9459960/domain_maskorig_ascii.txt)
[domain_meta.txt](https://github.com/NOAA-EMC/WW3/files/9459961/domain_meta.txt)
[domain_obstr_lev1.txt](https://github.com/NOAA-EMC/WW3/files/9459962/domain_obstr_lev1.txt)
Thanks in advance
Athira

**Reply by ukmo-ccbunney** [reply]
I see you have included a `ww3_grid.inp` file above. Previously you attached a `ww3_grid.nml` file.
Only one of those files is required. If you have the .nml file present, it will be read in preference to the .inp file...

**Reply by krishnanathira** [reply]
Yes. I am running the ww3_grid.inp file.

**Reply by ukmo-ccbunney** [reply]
OK - remove the ww3_grid.nml file otherwise the ww3_grid.inp file will be ignored.

I have run ww3_grid on your ww3_grid.inp file and it failed due to your input filenames being incorrect.
On line 19, you specify the filename as `domain.depth_ascii`, but the file is in fact called `domain_depth_ascii`.

I changed the input filenames on lines 19, 22 and 24 and the program ran successfully.

**Reply by ukmo-ccbunney** [reply]
i.e.:

```
-0.1 0.25 40 1. 3 1 '(....)' 'NAME' 'domain.depth_ascii'
$
$ Sub-grid information
 50  0.01  1  1 '(....)' 'NAME' 'domain.obstr_lev1'
$
 60  1  1 '(....)' 'NAME' 'domain.maskorig_ascii'
```

should be

```
-0.1 0.25 40 1. 3 1 '(....)' 'NAME' 'domain_depth_ascii'
$
$ Sub-grid information
 50  0.01  1  1 '(....)' 'NAME' 'domain_obstr_lev1'
$
 60  1  1 '(....)' 'NAME' 'domain_maskorig_ascii'
```

**Reply by krishnanathira** [reply]
Sorry, that was one typing error. grid.inp program is running successfully. later I ran my ww3_prnc.inp with the wind information and then ww3_multi.inp. Now the error again coming as

```
*** WAVEWATCH III ERROR IN WMINIT : *** 
     ERROR IN READING FROM INPUT FILE
     IOSTAT =    0
```

**Reply by ukmo-ccbunney** [reply]
That means you have a missing or malformed line in your ww3_multi.inp file...

**Reply by ukmo-ccbunney** [reply]
I recommend using the .nml files if possible - they don't have a prescribed format like the text .inp files so are more forgiving!

**Reply by krishnanathira** [reply]
Sure. Thanks a lot for your very helpful and prompt responses

**Reply by krishnanathira** [reply]
Hi
I have compiled the model with **Ifremer1** switch. Is that fine for running the model?

**Reply by ukmo-ccbunney** [reply]
> Hi I have compiled the model with **Ifremer1** switch. Is that fine for running the model?

Yes - that should be fine, if it is providing the functionality you require.
Your issue appears to be in the ww3_multi.inp file. Have you had any success using the ww3_multi.nml file instead?

**Reply by krishnanathira** [reply]
Sorry for the delayed response.
I have redone the runs with .nml files replacing .inp files and created mod.def and wind.ww3 using grid.nml and prnc.nml respectively. The model run has been done with shel.nml. Now the error is as follows,
```
Run ww3_shel on 
At line 1262 of file /home/ww3/WW3/model/src/ww3_shel.F90 (unit = 10, file = './ww3_shel.inp')
Fortran runtime error: Bad logical value while reading item 2
```

**Reply by ukmo-ccbunney** [reply]
That error message suggests you are still using the `.inp` file (ww3_shel.inp), rather than the `.nml` file.
Also - I am a bit confused - I thought you were running the multi-grid model driver `ww3_multi`, not `ww3_shel` - you mentioned the `ww3_multi.inp` file earlier as causing you problems...

Are you trying to run a **single grid** simulation? If so, you need to setup ww3_shel.nml correctly and run the ww3_shel executable.

If you are running a **multi-model simulation**, you need multiple "mod_def" files (one for each grid) and a ww3_multi.nml input file that is run with the ww3_multi executable.

**Reply by krishnanathira** [reply]
Thank you so much.
I was trying with a multi-grid in the beginning. But later switched to.NML files based runs as per your recommendation. I have used grid.nml, prnc.nml and shel.nml files for now. I understood that the errors were from the shell script which I was used to run the model. 
cp ww3_shel.nml.EA ww3_shel.inp instead of cp ww3_shel.nml.EA ww3_shel.nml
I have corrected it.
Thank you
